40.md

```diff
@@ -0,0 +1,46 @@
+NIP-40
+======
+
+Expiration Timestamp
+-----------------------------------
+
+`draft` `optional` `author:0xtlt`
+
+The `expiration` tag enables users to specify a timestamp at which the message should be considered expired and deleted by relays. This helps ensure that messages are not retained by the network after they are no longer relevant, which helps maintain the privacy of the messages. This is especially useful in cases where a replacement event is not available or feasible.
+
+#### Spec
+
+```
+tag: expiration
+options:
+ - [timestamp]: required
+```
+
+#### Example
+
+```json
+{
+    "pubkey": "<pub-key>",
+    "created_at": 1000000000,
+    "kind": 1,
+    "tags": [
+      ["expiration", 1600000000]
+    ],
+    "content": "This message will expire at the specified timestamp and be deleted by relays.\n",
+    "id": "<event-id>"
+}
+```
+
+Note: The timestamp should be in the same format as the created_at timestamp and should be interpreted as the time at which the message should be deleted by relays.
+
+Client Behavior
+---------------
+
+Clients SHOULD use the `supported_nips` field to learn if a relay supports this NIP. Clients SHOULD NOT send expiration events to relays that do not support this NIP.
+
+Suggested Use Cases
+-------------------
+
+* Temporary announcements - This tag can be used to make temporary announcements that are automatically deleted from the network after a certain amount of time. For example, an event organizer could use this tag to post announcements about an upcoming event, and the announcements will be automatically deleted once the event is over.
+* Limited-time offers - This tag can be used by businesses to make limited-time offers that expire after a certain amount of time. For example, a business could use this tag to make a special offer that is only available for a limited time, and the offer will be automatically deleted from the network once the time has passed.
+* Ephemeral messages - This tag can be used by individuals to send ephemeral messages that are automatically deleted from the network after a certain amount of time. For example, someone could use this tag to send a message that is only visible to the recipient for a short time, and the message will be automatically deleted once the time has passed. This can be useful for sending sensitive information that should not be retained on the network indefinitely.
```

Comment:
Don't broadcast sensitive information! Ever!

Reply:
So I am also responding here with the example of a signal-like application with custom relays

Reply:
In a group chat, in a DM, don't ever send information you require to not be available after a certain expiry date. It simply can't be enforced and Signal pretending it could is a disservice. With the very varied client landscape in nostr there is zero chance of all client agreeing to delete expired events. If information being linkable to your account after time X is a problem, don't sign and send it before time X neither.